internal/commands/server/k8s_e2e_test.go

```diff
@@ -1252,6 +1251,110 @@ func TestReferencePolicyLifecycle(t *testing.T) {
 			err = resources.Delete(ctx, gw)
 			require.NoError(t, err)
 
+			return ctx
+		}).
+		Assess("gateway controller watches reference policy changes", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			namespace := e2e.Namespace(ctx)
+			gatewayNamespace := namespace
+			gatewayName := envconf.RandomName("gw", 16)
+			certNamespace := envconf.RandomName("ns", 16)
+			certName := "consul-server-cert"
+			gatewayRefPolicyName := envconf.RandomName("refpolicy", 16)
+
+			resources := cfg.Client().Resources(namespace)
+
+			// Make a copy of the certificate Secret in a different namespace for the Gateway to reference.
+			// This is easier than creating the Gateway in a different namespace due to pre-installed ServiceAccount dependency.
+			certCopy := &core.Secret{}
+			require.NoError(t, resources.Get(ctx, certName, namespace, certCopy))
+			certCopy.SetNamespace(certNamespace)
+			certCopy.SetResourceVersion("")
+			require.NoError(t, resources.Create(ctx, &core.Namespace{ObjectMeta: meta.ObjectMeta{Name: certNamespace}}))
+			require.NoError(t, resources.Create(ctx, certCopy))
+
+			_, gc := createGatewayClass(ctx, t, resources)
+			require.Eventually(t, gatewayClassStatusCheck(ctx, resources, gc.Name, namespace, conditionAccepted), checkTimeout, checkInterval, "gatewayclass not accepted in the allotted time")
+
+			fromSelector := gateway.NamespacesFromAll
+
+			// Create a Gateway with a listener that has a CertificateRef to a different namespace
+			certNamespaceTyped := gateway.Namespace(certNamespace)
+			gw := createGateway(ctx, t, resources, gatewayName, gatewayNamespace, gc, []gateway.Listener{
+				{
+					Name:     "https",
+					Port:     gateway.PortNumber(e2e.HTTPReferencePolicyPort(ctx)),
+					Protocol: gateway.HTTPSProtocolType,
+					TLS: &gateway.GatewayTLSConfig{
+						CertificateRefs: []*gateway.SecretObjectReference{{
+							Name:      gateway.ObjectName(certName),
+							Namespace: &certNamespaceTyped,
+						}},
+					},
+					AllowedRoutes: &gateway.AllowedRoutes{
+						Namespaces: &gateway.RouteNamespaces{
+							From: &fromSelector,
+						},
+					},
```

Comment:
You're correct that there are no routes attaching. I kinda assumed things would break if we didn't have this minimal config here but didn't dig in to find out for sure.